src/control/control.cpp

```diff
@@ -102,27 +102,28 @@ void ControlDoublePrivate::insertAlias(const ConfigKey& alias, const ConfigKey&
 
 // static
 QSharedPointer<ControlDoublePrivate> ControlDoublePrivate::getControl(
-        const ConfigKey& key, bool warn, ControlObject* pCreatorCO,
-        bool bIgnoreNops, bool bTrack, bool bPersist, double defaultValue) {
-    if (key.isEmpty()) {
-        if (warn) {
-            qWarning() << "ControlDoublePrivate::getControl returning NULL"
-                       << "for empty ConfigKey.";
-        }
+        const ConfigKey& key,
+        ControlFlags flags,
+        ControlObject* pCreatorCO,
+        bool bIgnoreNops,
+        bool bTrack,
+        bool bPersist,
+        double defaultValue) {
+    VERIFY_OR_DEBUG_ASSERT(!key.isEmpty()) {
+        qWarning() << "ControlDoublePrivate::getControl returning NULL"
+                   << "for empty ConfigKey.";
         return QSharedPointer<ControlDoublePrivate>();
     }
 
-
     QSharedPointer<ControlDoublePrivate> pControl;
     // Scope for MMutexLocker.
     {
         MMutexLocker locker(&s_qCOHashMutex);
         auto it = s_qCOHash.constFind(key);
         if (it != s_qCOHash.constEnd()) {
             if (pCreatorCO) {
```

Comment:
This should work:
```
VERIFY_OR_DEBUG_ASSERT(!pCreateCO) {
} else {
}
```

Also use qWarning() instead of qDebug()

Reply:
I thought about this too, but then it instantly throws the debug assertion when `DEBUG_ASSERTIONS_FATAL` is enabled and does not print the log message before. That makes it harder to see which CO is responsible.